src/test/java/JsonParserTests.java

```diff
@@ -0,0 +1,83 @@
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.api.io.TempDir;
+import parser.JsonParser;
+import parser.NoSuchFileException;
+import shop.Cart;
+import shop.RealItem;
+import shop.VirtualItem;
+
+import java.io.File;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@Nested
+@DisplayName("Tests for JsonParser class")
+public class JsonParserTests {
+
+    private JsonParser jsonParser;
+    private Cart cart;
+    private final String FILE_PATH = "src/main/resources/TestCart.json";
+
+    @TempDir
+    public File tempDir;
+
+    @BeforeEach
+    public void setup() {
+        jsonParser = new JsonParser();
+        cart = new Cart("TestCart");
+        RealItem realItem = new RealItem();
+        realItem.setName("RealItem1");
+        realItem.setPrice(10.0);
+        realItem.setWeight(2.5);
+
+        VirtualItem virtualItem = new VirtualItem();
+        virtualItem.setName("VirtualItem1");
+        virtualItem.setPrice(20.0);
+        virtualItem.setSizeOnDisk(100.0);
+
+        cart.addRealItem(realItem);
+        cart.addVirtualItem(virtualItem);
+        cart.showItems();
+    }
+
+    @AfterEach
+    public void cleanup() {
+        new File(FILE_PATH).delete();
+    }
+
+    @Test
+    public void testWriteAndReadFromFile() {
```

Comment:
please divide this test into two: 
`testWriteToFile()` and `testReadFromFile()`. Each test will test the logic of its tested method